sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,165 @@
+}
+```
+
+You will notice that additional information is logged, like the Client Request ID of the operation.
+
+```shell
+Message: Azure.RequestFailedException : StatusCode: 404, ReasonPhrase: 'Not Found', Version: 1.1, Content: System.Net.Http.NoWriteNoSeekStreamContent, Headers:
+{
+  Connection: keep-alive
+  Date: Thu, 11 Apr 2019 00:16:57 GMT
+  Server: nginx/1.13.9
+  x-ms-client-request-id: cc49570c-9143-411e-a6c8-3287dd114034
+  x-ms-request-id: 2ad025f7-1fe8-4da0-8648-8290e774ed61
+  x-ms-correlation-request-id: 2ad025f7-1fe8-4da0-8648-8290e774ed61
+  Strict-Transport-Security: max-age=15724800; includeSubDomains;
+  Content-Length: 0
+}
+```
+
+### Enable client logging
+
+To troubleshoot issues with the library, first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+This library uses the standard [logging](https://docs.microsoft.com/dotnet/azure/sdk/logging) library. Basic information about HTTP sessions, such as URLs and headers, is logged at the `INFO` level.
+
+The simplest way to see the logs is to enable console logging. To create an Azure SDK log listener that outputs messages to the console, use the [AzureEventSourceListener.CreateConsoleLogger](https://docs.microsoft.com/dotnet/api/azure.core.diagnostics.azureeventsourcelistener.createconsolelogger?view=azure-dotnet) method:
+
+```csharp
+using Azure.Core.Diagnostics;
+
+// set up a listener to monitor logged events
+using AzureEventSourceListener listener = AzureEventSourceListener.CreateConsoleLogger();
+```
+
+#### Enable content logging
+
+By default only URI and headers are logged. To enable content logging, set the `Diagnostics.IsLoggingContentEnabled` client option. 
+
+``` c#
+ConfigurationClientOptions options = new ConfigurationClientOptions()
+{
+    Diagnostics =
+    {
+        IsLoggingContentEnabled = true
+    }
+};
+```
+#### Logging redacted headers and query parameters
+
+Some sensitive headers and query parameters are not logged by default and are displayed as "REDACTED", to include them in logs use the `Diagnostics.LoggedHeaderNames` and `Diagnostics.LoggedQueryParameters` client options.
+
+``` c#
+ConfigurationClientOptions options = new ConfigurationClientOptions()
+{
+    Diagnostics =
+    {
+        LoggedHeaderNames = { "x-ms-request-id" },
+        LoggedQueryParameters = { "api-version" }
+    }
+};
+```
+
+You can also disable redaction completely by adding a "*" to collections mentioned above.
+
+``` c#
+ConfigurationClientOptions options = new ConfigurationClientOptions()
+{
+    Diagnostics =
+    {
+        LoggedHeaderNames = { "*" },
+        LoggedQueryParameters = { "*" }
+    }
+};
+```
+
+To learn about other logging mechanisms, see [Azure SDK diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md).
```

Comment:
I wonder if we should combine this with the opening link so that the full context about logging is in one place.